src/javascript/class-scanner.ts

```diff
@@ -412,6 +605,24 @@ class ClassFinder implements Visitor {
 
     const warnings: Warning[] = [];
     const properties = extractPropertiesFromClass(astNode, this._document);
+    const methods = getMethods(astNode, this._document);
+
+    if (name) {
+      const protoFinder = new PrototypeMemberFinder(this._document, name);
+      this._document.visit([protoFinder]);
```

Comment:
this feels wrong to do. the finder does belong inside the class finder i think but i wasn't sure if `document.visit` is fine to use...

Reply:
This implementation seems like it's optimized to avoid creating unnecessary candidate properties, but I think the expensive part is going to be walking the AST, and creating candidate properties is super cheap.

I'd recommend instead creating a single PrototypeProperty finder that finds all such properties on all classes, and run that visitor at the same time as the other visitors in the Promise.all call at ~line 77.

Then we can match up the properties and methods with the found classes. Probably good to put the properties and methods in a map keyed by the class name since that's how we'll recognize them.

Reply:
that is exactly what i did in my first cut, but moved it to here :joy: 

i just thought it seemed a little confusing because a class could come after a prototype expression i suppose. or should we assume it never does? and just check `classes` for matching class names.

Reply:
to avoid having to check in both the finder and the class detection logic i mean